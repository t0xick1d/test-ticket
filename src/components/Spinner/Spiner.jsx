import React from 'react';

const Spinner = () => {
   return (
      <>
         <style>{`
        .spinner {
          border: 4px solid rgba(0, 0, 0, 0.1);
          width: 40px;
          height: 40px;
          border-radius: 50%;
          border-left-color: #09f;
          animation: spin 1s linear infinite;
          margin: auto;
        }

        @keyframes spin {
          to { transform: rotate(360deg); }
        }
      `}</style>
         <div className="spinner" />
      </>
   );
};

export default Spinner;
