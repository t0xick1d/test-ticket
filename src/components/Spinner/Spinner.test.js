import { render, screen } from '@testing-library/react';
import Spinner from './Spinner';

test('SPINNER TEST', () => {
   const { asFragment } = render(<Spinner />);
   expect(screen.getByTestId('containerdiv')).toBeInTheDocument();
   expect(screen.getByTestId('spindiv')).toBeInTheDocument();
   expect(asFragment()).toMatchSnapshot();
});
