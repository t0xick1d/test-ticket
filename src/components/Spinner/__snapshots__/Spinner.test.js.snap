// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SPINNER TEST 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="containerdiv"
  >
    <div
      class="spinner"
      data-testid="spindiv"
    />
  </div>
</DocumentFragment>
`;
